numpy/lib/function_base.py

```diff
@@ -3612,17 +3786,81 @@ def _quantile_is_valid(q):
     return True
 
 
-def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
-                           interpolation='linear', keepdims=False):
+def _validate_weights(a, q, axis, weights):
+    if weights is None:
+        wgt = None
+    else:
+        a = asanyarray(a)
+        wgt = np.asanyarray(weights)
+
+        if issubclass(a.dtype.type, (np.integer, np.bool_)):
+            result_dtype = np.result_type(a.dtype, wgt.dtype, 'f8')
+        else:
+            result_dtype = np.result_type(a.dtype, wgt.dtype)
+
+        broadcastable = False
+        if a.ndim == wgt.ndim and a.shape != wgt.shape:
```

Comment:
if `a.shape == wgt.shape`, then `broadcastable` should be `True`

Reply:
Good point